core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2644,6 +2638,15 @@ object ZIO extends ZIOFunctions {
     def toManaged: ZManaged[R, E, A] = ZManaged.fromAutoCloseable(io)
   }
 
+  implicit final class ZioRefineToOrDieOps[R, E <: Throwable, A](private val self: ZIO[R, E, A]) extends AnyVal {
+
+    /**
+     * Keeps some of the errors, and terminates the fiber with the rest.
+     */
+    final def refineToOrDie[E1 <: E: ClassTag](implicit ev: CanFail[E]): ZIO[R, E1, A] =
```

Comment:
Is the implicit class necessary? Does
```scala
final def refineToOrDie[E1 <: E: ClassTag](implicit ev1: E <:< Throwable, ev2: CanFail[E]): ZIO[R, E1, A] =
```
behave differently?

Reply:
That won't compile because the covariant `E` appears in contravariant position in the type bound, which is why we need the implicit class.

Reply:
Got it!